### PR TITLE
add support for ad3552r

### PR DESCRIPTION
**Pull request by spectrum70**
Add ad3552r driver, test script, example code, rst file and emu xml.

Tested:
- pytest
- invoke precommit
- pre-commit run --all-files
- executed sample on target

All tests passed.

Reviewed-by: Trevor Gamblin <tgamblin@baylibre.com>

# Description

Add support for ad3552r

## Type of change

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)
- [x ] This change requires a documentation update

# How has this been tested?

pytest -v --hw=ad3552r --emu --emu-xml=/home/angelo/dev-baylibre/adi/pyadi-iio/test/emu/devices/ad3552r.xml
test/test_ad3552r.py::test_ad3552r_raw_attr[raw-0-4000-1000-1-3-voltage0-adi.ad3552r] PASSED
test/test_ad3552r.py::test_ad3552r_raw_attr[raw-0-4000-1000-1-3-voltage1-adi.ad3552r] PASSED 

invoke precommit
Running supported_parts check
No parts missing from supported_parts.md

pre-commit run --all-files
codespell................................................................Passed
isort....................................................................Passed
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Debug Statements (Python)................................................Passed
Check docstring is first.................................................Passed
Flake8...................................................................Passed
black....................................................................Passed
blacken-docs.............................................................Passed
Insert license in comments...............................................Passed

example executed on target

**Test Configuration**:
* Hardware: Arch Linux, 6.5.0-rc2-devel-00345-gc2782531397f
* OS: Linux

# Documentation

rst files has been added accordingly

# Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ x] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: adi/ad3552r.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from adi.context_manager import context_manager
+from adi.rx_tx import tx
+
+
+class ad3552r(tx, context_manager):
+    """ AD3552R DAC """
+
+    _complex_data = False
+    _device_name = "AD3552R"
+    channel = []
```
Comment (tfcollins) [inline adi/ad3552r.py:17]: should be plural. channel -> channels

File: adi/ad3552r.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        """ Constructor for AD3552R driver class """
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        compatible_parts = [
+            "ad3552r",
+            "ad3542r",
+        ]
+
+        self._ctrl = None
```
[the remark this answers is not in the document]
Comment (spectrum70) [reply adi/ad3552r.py:29]: thanks, done

File: adi/ad3552r.py (hunk 1 above)
Comment (spectrum70) [reply adi/ad3552r.py:17]: done thankls

**Comment by spectrum70** [comment]
Hi @tfcollins , 
not clear how the PR acceptance process works, i don't see any reviewer assigned here, can you help me ?

File: adi/ad3552r.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+from adi.context_manager import context_manager
+from adi.rx_tx import tx
+
+
+class ad3552r(tx, context_manager):
+    """ AD3552R DAC """
+
+    _complex_data = False
+    _device_name = "AD3552R"
+    channels = []
```
Comment (tfcollins) [inline adi/ad3552r.py:17]: This is unused now. Is that expected? Not sure if this is just a name miss-match between channel and channel**s**

Comment (spectrum70) [reply adi/ad3552r.py]: Hi @tfcollins ,

fixed that, thanks.
Pushed a new version.

**Comment by spectrum70** [comment]
fixed conflicts with main

**Pull request merged**